### PR TITLE
Added Infinite ammo & clip.

### LINKED ISSUE #53: Unlimited Ammo and Clip
Hello sir is it possible if you add unlimited ammo and clip, please
thank you

### PR DESCRIPTION
**Pull request by xM4ddy**
Resolved Issue #53.

### CODE REVIEW

File: BigBaseV2/src/backend/looped/weapons/infinite_ammo.cpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#include "backend/looped/looped.hpp"
+#include "core/enums.hpp"
+#include "util/entity.hpp"
+#include "util/math.hpp"
+#include "util/notify.hpp"
```
Comment (Yimura) [inline BigBaseV2/src/backend/looped/weapons/infinite_ammo.cpp]: Remove unused includes.

File: BigBaseV2/src/backend/looped/weapons/infinite_mag.cpp [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#include "backend/looped/looped.hpp"
+#include "core/enums.hpp"
+#include "util/entity.hpp"
+#include "util/math.hpp"
+#include "util/notify.hpp"
```
Comment (Yimura) [inline BigBaseV2/src/backend/looped/weapons/infinite_mag.cpp]: Remove unused includes.

File: BigBaseV2/src/core/globals.hpp [hunk 3]
```diff
@@ -312,7 +316,9 @@ struct globals {
 			},
 			{
 				"weapons", {
-					{ "custom_weapon", (int)this->weapons.custom_weapon }
+					{ "custom_weapon", (int)this->weapons.custom_weapon },
+					{ "infinite_ammo", (int)this->weapons.infinite_ammo },
+					{ "infinite_mag", (int)this->weapons.infinite_mag }
```
Comment (Yimura) [inline BigBaseV2/src/core/globals.hpp]: Why cast this to int?
Boolean is a native type supported by json.

**Comment by Yimura** [comment]
Might also be cool to take a look at doing this in memory, this is something for later though since I don't think the offsets in the GTAV Classes have support for this yet.

**Review by ThatKakiGuy — APPROVED** [review]

**Review by Yimura — CHANGES_REQUESTED** [review]

**Comment by Yimura** [comment]
@xM4ddy please review the requested changes, otherwise I can't merge this in YimMenu.

File: BigBaseV2/src/core/globals.hpp [hunk 4]
```diff
@@ -203,6 +205,8 @@ struct globals {
 		this->vehicle.speedo_meter.y = j["vehicle"]["speedo_meter"]["position_y"];
 
 		this->weapons.custom_weapon = (CustomWeapon)j["weapons"]["custom_weapon"];
+		this->weapons.custom_weapon = j["weapons"]["infinite_ammo"];
+		this->weapons.custom_weapon = j["weapons"]["infinite_mag"];
```
Comment (Yimura) [inline BigBaseV2/src/core/globals.hpp:209]: You're assigning infinite_ammo and infinite_mag settings to the custom_weapon setting, this will not work.

Comment (Yimura) [inline BigBaseV2/src/core/globals.hpp]: You're assigning infinite_ammo and infinite_mag settings to the custom_weapon setting, this will not work.

**Review by Yimura — CHANGES_REQUESTED** [review]

**Review by Yimura — APPROVED** [review]

**Comment by Yimura** [comment]
@xM4ddy I've fixed the code this time but next time make sure you submit a fully working code base, you just made sure it compiled without actually fixing any problems. Doesn't appear either that you tried to think about what you did wrong and instead just made the compiler happy.

**Pull request merged**